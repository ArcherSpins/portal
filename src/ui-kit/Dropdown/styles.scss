
.drwn__wrap {
  .drwn__control {
    min-height: 2rem !important;
  }
}
