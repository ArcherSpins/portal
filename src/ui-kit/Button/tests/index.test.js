import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import React from 'react';

import { Button, ButtonWithProgress } from '..';
import styles from '../Button.module.scss';

describe('The Button', () => {
  it('default button renders correctly', () => {
    const button = <Button use="default" />;
    const wrapper = shallow(button);
    expect(wrapper.hasClass(styles.default)).toBeTruthy();
    const tree = renderer.create(button).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('transparent button renders correctly', () => {
    const button = <Button use="transparent" />;
    const wrapper = shallow(button);
    expect(wrapper.hasClass(styles.transparent)).toBeTruthy();
    const tree = renderer.create(button).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('grey button renders correctly', () => {
    const button = <Button use="grey" />;
    const wrapper = shallow(button);
    expect(wrapper.hasClass(styles.grey)).toBeTruthy();
    const tree = renderer.create(button).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call onClick', () => {
    const mockOnClick = jest.fn();
    const wrapper = shallow(<Button use="grey" onClick={mockOnClick} />);
    wrapper.find('button').simulate('click');
    expect(mockOnClick.mock.calls.length).toBe(1);
  });

  it('should show/hide child when changing loading prop', () => {
    const button = <ButtonWithProgress loading={false}>Some text</ButtonWithProgress>;
    const wrapper = shallow(button);
    expect(wrapper.find('[data-type="content"]').hasClass(styles.contentVisible)).toBeTruthy();
    wrapper.setProps({ loading: true });
    expect(wrapper.find('[data-type="content"]').hasClass(styles.contentHidden)).toBeTruthy();
  });

  it('should show/hide spinner when changing loading prop', () => {
    const button = <ButtonWithProgress loading={false}>Some text</ButtonWithProgress>;
    const wrapper = shallow(button);
    expect(wrapper.find('[data-type="spinner"]').hasClass(styles.contentHidden)).toBeTruthy();
    wrapper.setProps({ loading: true });
    expect(wrapper.find('[data-type="spinner"]').hasClass(styles.contentVisible)).toBeTruthy();
  });
});
