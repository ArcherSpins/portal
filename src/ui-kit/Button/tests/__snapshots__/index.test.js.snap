// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Button default button renders correctly 1`] = `
<button
  className="
        default
      "
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <span />
</button>
`;

exports[`The Button grey button renders correctly 1`] = `
<button
  className="
        grey
      "
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <span />
</button>
`;

exports[`The Button transparent button renders correctly 1`] = `
<button
  className="
        transparent
      "
  disabled={false}
  onClick={[Function]}
  type="button"
>
  <span />
</button>
`;
